sdcm/tester.py

```diff
@@ -1389,6 +1388,7 @@ def tearDown(self):
             self.log.exception('Exception in finalize_test method {}'.format(details))
             raise
         finally:
+            self.collect_logs()
```

Comment:
I think this might be o.k., 

but now if collect_logs would fail we might end up in a stuck job, since the stop_device_events() wouldn't happen. all those should take place not matter what happens.

let start using addCleanup as part of the the setUp 
https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup
and get rid of `finalize_test` completely

Reply:
collect_logs() should be called before clean_resources(), so I just moved collect_logs() back to finalize_test(). And use a try..finally to insure it's executed if stop_resources() raises exception.

Reply:
We have many tasks insider tearDown(), the tasks have fixed order. It's good idea to use Cleanup function.